src/main/java/org/wildfly/security/auth/spi/Attributes.java

```diff
@@ -44,44 +49,30 @@
      * Empty, read-only attribute collection.
      */
     Attributes EMPTY = new Attributes() {
+        @Override
         public Collection<Entry> entries() {
             return Collections.emptySet();
         }
 
+        @Override
         public int size(final String key) {
             return 0;
         }
 
-        public boolean remove(final String key) {
```

Comment:
This implementation does not need to be deleted.  Since the EMPTY Attributes is empty, a remove always can return false because there are no keys in it.